src/stores/chainTokenSelection/chainTokenSelection.tsx

```diff
@@ -0,0 +1,44 @@
+import type { ChainId } from '@lifi/types';
+import type { ChainTokenSelected } from '@lifi/widget';
+import { createWithEqualityFn } from 'zustand/traditional';
+
+interface OptionalChainTokenSelected {
+  chainId: ChainId | null | undefined;
+  tokenAddress: string | null | undefined;
+}
+
+interface ChainTokenSelectionState {
+  sourceChainToken: OptionalChainTokenSelected;
+  setSourceChainToken: (sourceChainToken: ChainTokenSelected) => void;
+  destinationChainToken: OptionalChainTokenSelected;
+  setDestinationChainToken: (destinationChainToken: ChainTokenSelected) => void;
+}
+
+const queryParameters = new URLSearchParams(window.location.search);
+const fromChain = queryParameters.get('fromChain');
+const fromToken = queryParameters.get('fromToken');
+const toChain = queryParameters.get('toChain');
+const toToken = queryParameters.get('toToken');
```

Comment:
Also, covering the URL-params to render that Solana alert on page refresh after a route has already been selected.